src/main/java/org/terasology/multiBlock2/system/MultiBlockServerSystem.java

```diff
@@ -385,10 +386,10 @@ private EntityRef createMultiBlockEntity(EntityRef mainBlockEntity, Vector3i mai
         return entityBuilder.build();
     }
 
-    private Region3i createAABB(Vector3i mainLocation, Collection<Vector3i> memberLocations) {
-        Region3i aabb = Region3i.createFromMinAndSize(mainLocation, new Vector3i(1, 1, 1));
+    private BlockRegion createRegion(Vector3i mainLocation, Collection<Vector3i> memberLocations) {
+        BlockRegion aabb = new BlockRegion(mainLocation);
         for (Vector3i memberLocation : memberLocations) {
-            aabb = aabb.expandToContain(memberLocation);
+            aabb = aabb.union(memberLocation);
         }
         return aabb;
     }
```

Comment:
@skaldarnar Correct me if I'm wrong but that sounds like a candidate for a stream and unioning?

Reply:
Yes, indeed. As we have a main location this simply becomes:

```java
return memberLocations.stream().reduce(new BlockRegion(mainLocation), BlockRegion::union, BlockRegion::union);
```